File: backend/text_generator.py
import random
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

# Add CORS middleware
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

TEXT_OPTIONS = [
    "Welcome to the Physical AI & Humanoid Robotics Textbook!",
    "Advanced robotics combines mechanical engineering, electrical engineering, and computer science.",
    "Machine learning algorithms enable robots to adapt to new environments.",
    "Humanoid robots represent one of the most challenging areas in robotics research.",
    "Kinematics and dynamics are fundamental concepts in robot motion planning.",
    "Artificial intelligence is revolutionizing the capabilities of autonomous systems.",
    "Sensor fusion allows robots to perceive their environment accurately.",
    "Control theory provides the mathematical foundation for robotic movement.",
    "Computer vision enables robots to interpret visual information from cameras.",
    "Natural language processing allows humans to interact with robots using speech."
]

@app.get("/")
def read_root():
    return {"message": "Text Generator API is running on port 3000!"}

@app.get("/generate-text")
def generate_text():
    """Generate a random text related to robotics and AI"""
    text = random.choice(TEXT_OPTIONS)
    return {"text": text}

@app.get("/generate-text-multiple")
def generate_text_multiple(count: int = 5):
    """Generate multiple random texts related to robotics and AI"""
    texts = [random.choice(TEXT_OPTIONS) for _ in range(min(count, 10))]  # Limit to 10
    return {"texts": texts}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("text_generator:app", host="0.0.0.0", port=3000, reload=True)

File: text_generator.js
const express = require('express');
const cors = require('cors');
const app = express();
const PORT = 3000;

// Enable CORS for all routes
app.use(cors());

// Sample text options
const TEXT_OPTIONS = [
  "Welcome to the Physical AI & Humanoid Robotics Textbook!",
  "Advanced robotics combines mechanical engineering, electrical engineering, and computer science.",
  "Machine learning algorithms enable robots to adapt to new environments.",
  "Humanoid robots represent one of the most challenging areas in robotics research.",
  "Kinematics and dynamics are fundamental concepts in robot motion planning.",
  "Artificial intelligence is revolutionizing the capabilities of autonomous systems.",
  "Sensor fusion allows robots to perceive their environment accurately.",
  "Control theory provides the mathematical foundation for robotic movement.",
  "Computer vision enables robots to interpret visual information from cameras.",
  "Natural language processing allows humans to interact with robots using speech."
];

// Root endpoint
app.get('/', (req, res) => {
  res.json({ message: "Text Generator API is running on port 3000!" });
});

// Generate single text endpoint
app.get('/generate-text', (req, res) => {
  const randomIndex = Math.floor(Math.random() * TEXT_OPTIONS.length);
  const text = TEXT_OPTIONS[randomIndex];
  res.json({ text });
});

// Generate multiple texts endpoint
app.get('/generate-text-multiple', (req, res) => {
  const count = parseInt(req.query.count) || 5;
  const validCount = Math.min(Math.max(count, 1), 10); // Between 1 and 10
  const texts = [];
  
  for (let i = 0; i < validCount; i++) {
    const randomIndex = Math.floor(Math.random() * TEXT_OPTIONS.length);
    texts.push(TEXT_OPTIONS[randomIndex]);
  }
  
  res.json({ texts });
});

app.listen(PORT, '0.0.0.0', () => {
  console.log(`Text Generator server running on port ${PORT}`);
});